.github/workflows/main.yml

```diff
@@ -0,0 +1,134 @@
+name: CI Pipeline
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'python-flask/**'
+      - 'c-sharp-razor/**'
+      - 'rust-actix/**'
+      - 'ruby-sinatra/**'
+      - 'javascript-express/**'
+      - 'go-gin/**'
+      - 'go-gorilla/**'
+      - '/**'
+
+jobs:
+  check-root-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      root_changed: ${{ steps.set-flags.outputs.root_changed }}
+      service_changed: ${{ steps.set-flags.outputs.service_changed }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check if root files or any service changed
+        id: set-flags
+        run: |
+          # Initialize flags
+          root_changed="false"
+          any_service_changed="false"
+
+          # Get the list of changed files by comparing against the base branch
+          git fetch origin main
+          changed_files=$(git diff --name-only origin/main...HEAD)
+
+          # Check if any of the changed files are outside the known service directories
+          for file in $changed_files; do
+            if [[ ! "$file" =~ ^(python-flask/|c-sharp-razor/|rust-actix/|ruby-sinatra/|javascript-express/|go-gin/|go-gorilla/|\.github/) ]]; then
+              root_changed="true"
+              echo "Root-level file changed: $file"
+            fi
+            if [[ "$file" =~ ^(python-flask/|c-sharp-razor/|rust-actix/|ruby-sinatra/|javascript-express/|go-gin/|go-gorilla/) ]]; then
+              any_service_changed="true"
+              echo "Service-level file changed: $file"
+            fi
+          done
+
+          # Set outputs
+          echo "::set-output name=root_changed::$root_changed"
```

Comment:
I don't get this command here. You are echo to create a root_changed and a service changed in the environment?

Reply:
it is a bit weird but if it works it works

Reply:
Me too. I tried to do it with env-vars within the workflow but they are job-local so it it took too much time to configure so I decided on outputs instead.